### PR TITLE
Refactor migration logic with comprehensive error handling and test coverage

### PR DESCRIPTION
**Pull request by MiaKoring**
## Overview
This PR refactors migration handling to ensure all models are accounted for during migrations, introduces comprehensive, recoverable error handling, and adds extensive tests for migration edge cases.

### Core migration logic (ModelContainer.swift)
- migrate now runs migration stages in a loop inside a single transaction, applying multiple complex stages until no further migrations are needed.
- After each stage:
  - Runs the stage's migrationBlock
  - Validates remaining tables via unmigratedSchemas(from:) and throws ManagedObjectContextError.modelsUnhandledAfterMigration if any old-schema rows remain
  - Calls context.cleanupOldSchema(originVersion) to delete old tables
  - Calls didFinishMigration?(context)
- determineMigrationStage now throws explicit errors instead of fatalError:
  - emptySchemaMigrationPlan(migration) when migration.schemas is empty
  - noSchemaMatchingVersion(migration, version) when the DB version has no matching registered schema
  - noMigrationForOutdatedModelVersion(migration, version) when no migration stage starts from the current DB version

### Migration helpers and visibility (ManagedObjectContext+Migration.swift & ManagedObjectContext.swift)
- cleanupOldSchema visibility reduced from public to package and is guarded by isInActiveMigration.
- Added ManagedObjectContext.getNonEmptySchemas() -> [String] to list non-empty tables excluding MigrationTable.schema and sqlite_ prefixed tables.
- getAllStoredSchemas() updated to filter out sqlite_ prefixed tables (unified filtering).
- Various helper functions for creating/renaming/deleting tables and recording migrations retained in migration extension.

### Error model expansion (ManagedObjectContextError.swift)
- Added four migration-specific error cases with localized descriptions, failure reasons, and recovery suggestions:
  - modelsUnhandledAfterMigration(origin, destination, [String])
  - emptySchemaMigrationPlan(SchemaMigrationPlan.Type)
  - noSchemaMatchingVersion(SchemaMigrationPlan.Type, ModelVersion)
  - noMigrationForOutdatedModelVersion(SchemaMigrationPlan.Type, ModelVersion)

### API surface and protocol changes
- Date.sqliteFormatStyle visibility changed from package to public.
- VersionedSchema protocol now conforms to Sendable.
- cleanupOldSchema access reduced to package (see above).

### Tests — expanded coverage for migration error paths
- New/updated tests added:
  - EmptySchemaMigrationPlan — asserts emptySchemaMigrationPlan is thrown for empty schemas in a plan
  - NoMigrationForOutdatedModelVersion — asserts noMigrationForOutdatedModelVersion is thrown when migration stages are missing
  - NoSchemaMatchingVersion — asserts noSchemaMatchingVersion is thrown when DB version lacks a registered schema
  - ModelsUnhandledAfterMigration — asserts modelsUnhandledAfterMigration is thrown when migration leaves rows unhandled
  - Unchanged+Delete — extended migration plan with additional stages (V0_0_2→V0_0_3, V0_0_3→V0_0_4) and corresponding assertions

## Highlights (notable improvements)
- Replacing fatalError paths with explicit ManagedObjectContextError cases improves robustness and makes migration failures recoverable and diagnosable.
- Thorough unit tests for previously unhandled edge cases significantly raise confidence in migration behavior.
- Processing multiple migration stages within a single transaction improves atomicity and consistency of multi-step migrations.

## Impact
- Stronger defensive behavior during migrations (clear, actionable errors instead of crashes).
- Better detection and prevention of incomplete migrations via unmigratedSchemas checks.
- Minor API visibility and protocol updates (cleanupOldSchema visibility, Date formatting accessor, VersionedSchema Sendable conformance) that may affect external consumers.

### CODE REVIEW

File: Tests/VeinTests/Migrations/Unchanged+Delete.swift [hunk 1]
```diff
@@ -339,5 +345,38 @@ fileprivate enum SimpleMigrationSuccess: SchemaMigrationPlan {
...
+            
+            for model in models {
+                let date = Date(timeIntervalSince1970: Double(model.date))
+                let newModel = SimpleSchemaV0_0_4.Test(addedAt: date)
+                try context.insert(newModel)
+                try context.delete(model)
+            }
+        },
+        didMigrate: nil
+    )
```
[the remark this answers is not in the document]
Comment (MiaKoring) [reply Tests/VeinTests/Migrations/Unchanged+Delete.swift:377]: context.insert() catches noSuchTable and creates it if necessary using 
PersistentModel/migrate(in: ManagedObjectContext):
```swift
internal func migrate(in context: Vein.ManagedObjectContext) throws(ManagedObjectContextError) {
        var builder = context.createSchema(Self.schema)
            .id()
        
        // dropping first to not create `id` twice
        for field in _fields.dropFirst() {
            field.migrate(on: &builder)
        }
        
        do {
            try builder.run()
            try context.registerMigration(schema: _getSchema(), version: Self.version)
        } catch let error as ManagedObjectContextError {
            throw error
        } catch let error as SQLite.Result {
            throw error.parse()
        } catch { throw .other(message: error.localizedDescription) }
        
    }
    ```

[the remark this answers is not in the document]
Comment (MiaKoring) [reply Tests/VeinTests/Migrations/Unchanged+Delete.swift:377]: @coderabbitai please add this trait to your learnings
